### PR TITLE
Implement global config .config/autoimport/config.toml

### PR DESCRIPTION
**Pull request by Jasha10**
Inspired by the prior art in PR #172, this PR implements discovery and loading of a global config file for `autoimport`. The global config is located at `$XDG_CONFIG_HOME/autoimport/config.toml`.

## Checklist

* [X] Add test cases to all the changes you introduce
* [X] Update the documentation for the changes

### CODE REVIEW

File: tests/e2e/test_cli.py [hunk 1]
```diff
@@ -140,6 +141,82 @@ def test_pyproject_common_statements(runner: CliRunner, tmpdir: LocalPath) -> No
...
+            True, True, False, "from g import G\nfrom l import L", id="global-and-local"
+        ),
+        pytest.param(
+            True,
+            False,
+            True,
+            "from g import G\nfrom p import P",
+            id="global-and-pyproject",
+        ),
+        pytest.param(False, True, True, "from l import L", id="local-and-pyproject"),
```
Comment (lyz-code) [inline tests/e2e/test_cli.py:160]: Shouldn't this be `from l import L\nfrom p import P`?

File: tests/e2e/test_cli.py [hunk 2]
```diff
@@ -140,6 +141,82 @@ def test_pyproject_common_statements(runner: CliRunner, tmpdir: LocalPath) -> No
...
+        pytest.param(
+            True,
+            True,
+            True,
+            "from g import G\nfrom l import L",
+            id="global-and-local-and-pyproject",
```
Comment (lyz-code) [inline tests/e2e/test_cli.py:166]: Shouldn't this be `from g import G\nfrom l import L\nfrom p import P`?

File: tests/e2e/test_cli.py [hunk 3]
```diff
@@ -140,6 +141,82 @@ def test_pyproject_common_statements(runner: CliRunner, tmpdir: LocalPath) -> No
...
+        pyproject_path = tmpdir / "pyproject.toml"
+        pyproject_path.write(conf_pyproject)
+    with tmpdir.as_cwd():
+
+        result = runner.invoke(cli, args, env=env)
+
+    assert result.exit_code == 0
+    assert code_path.read() == expected_imports + "\n" + original_code
+
+
```
Comment (lyz-code) [inline tests/e2e/test_cli.py:219]: I'm missing a test that makes sure that the configuration overwriting works, say that the global sets `from g import G` and the local sets `from g import L`, the latter should be used. If you think we can delegate it to `maison` I'm fine with that too

**Review by lyz-code — CHANGES_REQUESTED** [review]
Wow thank you so much for the contribution, I loved how you made everything in one test.

File: tests/e2e/test_cli.py (hunk 1 above)
Comment (Jasha10) [reply tests/e2e/test_cli.py]: Updated in https://github.com/lyz-code/autoimport/pull/206/commits/f22c4a13dc71d79aee7cc67eaa97007108279028.

File: tests/e2e/test_cli.py (hunk 2 above)
Comment (Jasha10) [reply tests/e2e/test_cli.py:172]: Updated in https://github.com/lyz-code/autoimport/pull/206/commits/f22c4a13dc71d79aee7cc67eaa97007108279028.

File: tests/e2e/test_cli.py (hunk 3 above)
Comment (Jasha10) [reply tests/e2e/test_cli.py:225]: I've added the test in https://github.com/lyz-code/autoimport/pull/206/commits/f7b71b465ce6c22d4f2e4ad88fd2fd953bf812d0.

File: tests/e2e/test_cli.py (hunk 1 above)
Comment (lyz-code) [reply tests/e2e/test_cli.py]: Lovely, thanks!

File: tests/e2e/test_cli.py (hunk 2 above)
Comment (lyz-code) [reply tests/e2e/test_cli.py:172]: Lovely, thanks!

File: tests/e2e/test_cli.py (hunk 3 above)
Comment (lyz-code) [reply tests/e2e/test_cli.py:225]: Lovely, thanks!

**Pull request merged**